src/main/java/org/cascadebot/cascadebot/utils/diff/DiffUtils.java

```diff
@@ -0,0 +1,76 @@
+package org.cascadebot.cascadebot.utils.diff;
+
+import com.google.gson.reflect.TypeToken;
+import lombok.SneakyThrows;
+import org.cascadebot.cascadebot.CascadeBot;
+import org.cascadebot.cascadebot.utils.lists.CollectionDiff;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DiffUtils {
+
+    public static Difference diff(Object oldObj, Object newObj) {
+        return diff(new Difference(), "", oldObj, newObj);
+    }
+
+    @SneakyThrows
+    private static Difference diff(Difference currentDiff, String path, Object oldObj, Object newObj) {
+        Type mapType = new TypeToken<Map<String, Object>>() {
+        }.getType();
+        Map<String, Object> oldMap = CascadeBot.getGSON().fromJson(CascadeBot.getGSON().toJson(oldObj), mapType);
+        Map<String, Object> newMap = CascadeBot.getGSON().fromJson(CascadeBot.getGSON().toJson(newObj), mapType);
```

Comment:
Why are we using JSON here instead of reflection?

Reply:
Because it's easier, also I'm running into some reflection issues where java isn't finding all the fields, might have something to do with kotlin interopt. Might look into switching over the reflection in the future.

Reply:
Although switching over to reflection would resolve this https://github.com/CascadeBot/CascadeBot/pull/254#discussion_r642065697